Issue #432 - work in progress

petemoore opened the pull request:
Initially using the AWS CLI to generate the mappings, we could consider using the python client instead.

We might also want to filter just on the availability zones that are referenced and/or the instance types that are referenced, in order to have a smaller footprint - although having everything can be useful for tracking changes over time, and also maybe keeps things more simple.

jwhitlock commented:
I've added code to `generate/workers.py` to use these JSON files to dynamically filter non-available worker types by zone. I've also added an assertion that at least one launch config is generated, to avoid pools with no launch configurations. I could add a second check that each region (like "us-west-1") also has at least one launch config.

petemoore commented:
> I've added code to `generate/workers.py` to use these JSON files to dynamically filter non-available worker types by zone. I've also added an assertion that at least one launch config is generated, to avoid pools with no launch configurations. I could add a second check that each region (like "us-west-1") also has at least one launch config.

This is awesome, many thanks John! I noticed that `m5.2xlarge` is only supported in two availability zones of the regions worker-manager uses (`us-west-1`, `us-west-2`, `us-east-1`), with both of them being in `us-west-1` (`us-west-1a` and `us-west-1b`). However there are several other regions that do support this instance type, that Worker Manager doesn't use:

```
ap-northeast-1
ap-northeast-2
ap-northeast-3
ap-south-1
ap-southeast-1
ap-southeast-2
ca-central-1
eu-central-1
eu-north-1
eu-west-1
eu-west-2
eu-west-3
sa-east-1
us-east-2
```

I'm not sure why Worker Manager only uses `us-east-1`, `us-west-1` and `us-west-2` but perhaps we should review that decision.

petemoore commented:
@jwhitlock I've renamed things a little, cleaned up a typo, created new commits, and added the google cloud machine types to the `/config` directory. The format is a little different from the amazon instance types; there is a single file rather than one per zone. The data is structured as a list of (machine type, zone) pairs (which was easier to create than e.g. a mapping from zone to machine types). Do you think this is ok? We still need to update [`machine_in_zone(machineType, zone)`](https://github.com/mozilla/community-tc-config/blob/266a6c02d7d3e6bf86bd192a986355c18266fb23/generate/workers.py#L269-L278) to use this new data source.

petemoore commented:
Thanks for your commits John! I don't think I can approve the PR since I created it, but I'm happy to confirm here in writing that I r+ your commits! Hopefully GitHub still allows to review the PR even though you have contributed some of the commits. If not we'll have to rope somebody else in.
